bokeh/sphinxext/bokehjs_script.py

```diff
@@ -0,0 +1,151 @@
+not provided, all BokehJS files will be included in the script tags:
+
+.. code-block:: rest
+
+    .. bokehjs-script-tag::
+        :js_files: bokeh-widgets,bokeh-tables
+
+This directive is identical to the standard ``code-block`` directive
+that Sphinx supplies, with the addition of one new option:
+
+js_files (string):
+    A comma separated string of JavaScript files to look for.
+
+Examples
+--------
+
+The inline example code above produces the following output:
+
+.. bokehjs-script-tag::
+    :js_files: bokeh-widgets,bokeh-tables
+
+'''
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+from os.path import basename
+
+# External imports
+from docutils import nodes
+from docutils.parsers.rst.directives import unchanged
+from sphinx.directives.code import CodeBlock
+
+# Bokeh imports
+from .templates import BJS_SCRIPT_TAGS
+from .util import get_codeblock_node, get_sphinx_resources
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    'bokehjs_script',
+    'BokehJSScript',
+    'html_depart_bokehjs_script',
+    'html_visit_bokehjs_script',
+    'setup',
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# Dev API
+#-----------------------------------------------------------------------------
+
+class bokehjs_script(nodes.General, nodes.Element):
+    pass
+
+
+class BokehJSScript(CodeBlock):
+
+    option_spec = CodeBlock.option_spec
+    option_spec.update(js_files=unchanged)
+
+    def run(self):
+        env = self.state.document.settings.env
+
+        rst_source = self.state_machine.node.document['source']
+        rst_filename = basename(rst_source)
+
+        target_id = "%s.ccb-%d" % (rst_filename, env.new_serialno('ccb'))
+        target_id = target_id.replace(".", "-")
+        target_node = nodes.target('', '', ids=[target_id])
+
+        node = bokehjs_script()
+        node['target_id'] = target_id
+
+        resources = get_sphinx_resources(include_bokehjs_api=False)
+
+        js_components = self.options.get('js_files', 'all')
+        if js_components != 'all':
+            resources = self.get_modified_resources(js_components, resources)
+
+        html_source = BJS_SCRIPT_TAGS.render(
+            js_files=resources.js_files,
+            hashes=resources.hashes)
```

Comment:
how does this perform when building docs from a dev install (or for a dev build or rc), where there are no hashes?

Reply:
Are you referring to the time it takes to build when running the `make` command?

I see there are `{%- if hashes %} ... {%- endif -%}` checks in two other templates so this should probably be included in this template too.

Reply:
So running `BOKEH_DOCS_CDN=local make clean html serve` loaded the following for me in the embed guide without the hashes check:
```html
<script src="/en/latest/static/js/bokeh.min.js"></script>
<script src="/en/latest/static/js/bokeh-widgets.min.js"></script>
<script src="/en/latest/static/js/bokeh-tables.min.js"></script>
```

Added the hashes check just to be safe.